app/src/main/java/org/mozilla/focus/settings/SettingsFragment.java

```diff
@@ -76,6 +100,10 @@ public void onPause() {
         getPreferenceManager().getSharedPreferences().unregisterOnSharedPreferenceChangeListener(this);
     }
 
+    public void setTitleUpdater(TitleUpdater titleUpdater) {
+        this.titleUpdater = titleUpdater;
+    }
```

Comment:
`this.titleUpdater` will be null if the `Fragment` gets destroyed while the app is in the background and the next time you resume the app. Android will re-create the fragment but no one is calling `setTitleUpdater()` on it. Only the arguments `Bundle` will be preserved and passed to the new instance again.

As the `TitleUpdater` is in fact the activity of the fragment, you can remove this reference completely and just cast `getActivity()` to `TitleUpdater` where needed (Watch out, getActivity() could be null). If you want to be explicit about the activity having to implement this interface then you could override onAttach() and throw if it isn't an `instanceof TitleUpdater`.